charts/osm/crds/multiclusterservice.yaml

```diff
@@ -0,0 +1,63 @@
+# Custom Resource Definition (CRD) for OSM's multicluster config specification.
+#
+# Copyright Open Service Mesh authors.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: multiclusterservices.config.openservicemesh.io
+spec:
+  group: config.openservicemesh.io
+  scope: Namespaced # osm-system is the required namespace
+  names:
+    kind: MultiClusterService
```

Comment:
I am not sure if the right name for this crd would be `MultiClusterService`, since there is no mention of a service anywhere in the crd, we might have to come up with a different name that is a better representation what the crd entails

Reply:
@steeling maybe something like `MultiClusterConfiguration`?

Reply:
I think MultiClusterService is fitting. It really is closer to a service.

FWIW `Service` has no mention of service either, it's just in the name, and generates a service in K8s. Similarly, the MultiClusterService generates (multiple) services, but in OSM (service.MeshService).

From a user perspective, a MultiClusterService of name `bookstore` with namespace `bookstore`, and clusters `cluster-x` and `cluster-y`, will generate 3 services:

`bookstore.bookstore.svc.cluster.global`
`bookstore.bookstore.svc.cluster.cluster-x`
`bookstore.bookstore.svc.cluster.cluster-y`

Reply:
This is true only if the service and service account name is the same, which might not always be the case. I think I prefer `MultiClusterConfiguration`

Reply:
That's actually not how this will work out. Since this refers to multi cluster services (where it doesn't have the ability to extract the identity from the pods, or services), it infers the service account from this object.

It *does* create several service.MeshService. Moreover, a MultiClusterConfiguration sounds like a singleton for a multicluster, whereas there will be 1 of these *per service*.

Reply:
Post an offline discussion we decided to keep the name as is, let's add a comment to mention that the same of this resource needs to match the name of the service this resource intends to target